src/librekey/key_store_g10.c

```diff
@@ -779,14 +1164,218 @@ write_seckey(s_exp_t *s_exp, pgp_seckey_t *key)
     return true;
 }
 
+static bool
+write_protected_seckey(s_exp_t *s_exp, pgp_seckey_t *key, const uint8_t *passphrase)
+{
+    const format_info *format;
+    s_exp_t            raw_s_exp = {0};
+    s_exp_t *          sub_s_exp, *sub_sub_s_exp, *sub_sub_sub_s_exp;
+    botan_cipher_t     encrypt = NULL;
+    uint8_t            derived_key[PGP_MAX_KEY_SIZE];
+    unsigned           keysize;
+    pgp_memory_t       raw = {0};
+    time_t             now;
+    size_t             output_written = 0;
+    size_t             input_consumed = 0;
+
+    if (key->s2k_specifier != PGP_S2KS_ITERATED_AND_SALTED) {
+        fprintf(stderr, "s2k should be iterated and salted\n");
+        return false;
+    }
+
+    // if we had encrypted block, don't renecrypt
+    if (key->encrypted_len > 0 && key->encrypted != NULL) {
+        goto write;
```

Comment:
Yes, you're right. I rebased this commit after first push, but second push was without `-f`. So... github declined it.

Anyway, I pushed one more time and it should be fixed.